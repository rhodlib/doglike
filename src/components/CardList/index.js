import React from "react";
import PropTypes from "prop-types";
import Card from "../Card";
import styles from "./CardList.module.css";

/**
 * Component that make a list of button's.
 */
export const CardList = ({ dogList = [] }) => {
  return (
    <div className={styles.CardList}>
      {dogList.map(dog => (
        <Card key={dog} dogName={dog} />
      ))}
    </div>
  );
};

CardList.propTypes = {
  dogList: PropTypes.arrayOf(PropTypes.string)
};

export default CardList;
