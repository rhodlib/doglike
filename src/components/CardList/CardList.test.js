import React from "react";
import { render } from "@testing-library/react";
import CardList from "./index";
import { BrowserRouter as Router } from "react-router-dom";

it("Matches snapshot", () => {
  const { asFragment } = render(
    <Router>
      <CardList dogList={["test"]} />
    </Router>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("Without doglist", () => {
  const { asFragment } = render(
    <Router>
      <CardList />
    </Router>
  );
  expect(asFragment()).toMatchSnapshot();
});
