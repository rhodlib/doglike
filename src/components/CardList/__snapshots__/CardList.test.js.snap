// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="CardList"
  >
    <a
      class="Card"
      href="/post/test"
    >
      <img
        alt="paw"
        class="CardImage"
        src="blackpaw.png"
      />
      <p
        class="CardNameParagraph"
      >
        test
      </p>
    </a>
  </div>
</DocumentFragment>
`;

exports[`Without doglist 1`] = `
<DocumentFragment>
  <div
    class="CardList"
  />
</DocumentFragment>
`;
